master/pkg/searcher/operations.go

```diff
@@ -168,7 +162,8 @@ func (c Checkpoint) String() string {
 	return fmt.Sprintf("{Checkpoint %s}", c.RequestID)
 }
 
-// Train is an operation emitted by search methods to signal the trial train for a specified length.
+// Train is an operation emitted by search methods to signal the trial train until
+// its total batches trained equals the specified length.
 type Train struct {
```

Comment:
Might be worth renaming the "Train" operation which ends with a "validaitonCompleted".  Naming-wise it looks more like we took a more full-featured system and deleted things in a way that resulted in these naming convetions, rather than that we chose the best names for each object.  And if we don't rename it now I doubt we ever do...

Reply:
Yeah ok, hmm. Could call it `TrainUntilAndValidate`? I mean, that's exactly what it is.

Reply:
Maybe `ValidateAt` or `ValidateAfter`?